### DISCUSSION TITLE
Access grid information from a vertex-grid (disV)

### DISCUSSION THREAD

**Discussion by d-strom** [modflowpy/flopy#1992] [Q&A] [answered]
Hi,

I have browsed the forum and the flopy documentation to no avail... 

I'm trying to access the grid information of a pre-generated vertex grid (by modelmuse).

I figured the flopy.utils.gridgen.get_gridprops_disv() should be useful for this, but for whatever reason I'm not succeeding in importing the grid properties.

I did try to follow the steps in this example:
https://flopy.readthedocs.io/en/latest/Notebooks/modelgrid_examples.html

From what it seems it needs to generate or import a normal dis-grid and insert a refinement after.
But what if i want to access the modelgrid class and the grid properties/domain directly from a .disv-file generated outside of the code?

Can it be done?

edit: By the way. Running the following code

`# load a modflow-6 simulation
sim = flopy.mf6.MFSimulation.load(sim_ws=tmp_d, verbosity_level=0)

# get a model object from the simulation
m = sim.get_model("mod0000")

# access the modelgrid
modelgrid = m.modelgrid()`

gives me this error 

> ---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[10], line 8
      5 m = sim.get_model("mod0000")
      7 # access the modelgrid
----> 8 modelgrid = m.modelgrid()
     10 m.dis.top.plot()
  

AttributeError: 'NoneType' object has no attribute 'modelgrid'

> 

Sincerely
David

**Comment by wpbonelli** [comment]
You can load a vertex grid [directly from a binary grid file](https://github.com/modflowpy/flopy/blob/develop/flopy/discretization/vertexgrid.py#L550). Alternatively, you can load a simulation with `MFSimulation.load()` per usual, then retrieve the modelgrid from a model as you describe in your edit.

The error you are seeing indicates no model was found with the provided name. In this case flopy returns `None` from `get_model()`.

**Reply by d-strom** [reply]
Thanks for the advice.

When I try to access the information with the classmethod described i get the following error:

![image](https://github.com/modflowpy/flopy/assets/74509984/61a80362-f63c-431a-bbcf-297b31f3f8b8)


![image](https://github.com/modflowpy/flopy/assets/74509984/a04cbd06-b16f-4c34-817c-14e59c08d0c4)

Any idea of whats going on? I have tried with a "known" working disv-grid from the examples to make sure its not my input-file that is the problem.

**Reply by wpbonelli** [reply, marked as answer]
Without your grid file it's hard to diagnose. It looks like it's not a DISV grid but a regular DIS. Note the same `from_binary_grid_file()` method is available on `StructuredGrid` as well, does that work?

**Reply by d-strom** [reply]
Turns out it was using the non-updated file from a temp dir. Seems to be working now! Thanks!
